apps/api/src/app/shared/helpers/verify-payload.service.ts

```diff
@@ -0,0 +1,82 @@
+import { TemplateSystemVariables, ITemplateVariable } from '@novu/shared';
+import { ApiException } from '../exceptions/api.exception';
+
+export class VerifyPayloadService {
+  checkRequired(variables: ITemplateVariable[], payload: Record<string, unknown>): string[] {
+    const invalidKeys: string[] = [];
+
+    for (const variable of variables.filter((vari) => vari.required && !this.isSystemVariable(vari.name))) {
+      let value;
+
+      try {
+        value = variable.name.split('.').reduce((a, b) => a[b], payload);
+      } catch (e) {
+        value = null;
+      }
+
+      const variableTypeHumanize = {
+        String: 'Value',
+        Array: 'Array',
+        Boolean: 'Boolean',
+      }[variable.type];
```

Comment:
I understand that you are trying to __humanize the variable type__. So just swap the order in the name would reads more natural I think: `humanizedVariableType`. Though you could also to refer to it as `variableTypeLabel` that might go more in line with being the text will be rendered like in an UI.

Reply:
Sure, its an old code, I just moved it.